server/controllers/authController.js

```diff
@@ -46,5 +48,44 @@ export const signup = async (req, res) => {
 };
 
 export const signin = async (req, res) => {
-
+  try {
+    const { email, password } = req.body;
+    const matchedEmailContent = await database.findUser({ email, password });
+    const emailFound = matchedEmailContent.rows.length;
+    if (emailFound) {
+      const hashedPassword = matchedEmailContent.rows[0].password;
+      const pswdMatches = compareToHashed(hashedPassword, password);
+      if (pswdMatches) {
+        const { firstname, lastname, id } = matchedEmailContent.rows[0];
+        const newToken = generateToken({ id });
+        res.status(200).json({
+          status: res.statusCode,
+          message: 'User logged in successfully',
+          data: {
+            token: newToken,
+            userDetails: {
+              FirstName: firstname,
+              LastName: lastname,
+              Email: email,
+            },
+          },
+        });
+      } else {
+        res.status(401).json({
+          status: res.statusCode,
+          error: 'Invalid credentials',
+        });
+      }
+    } else {
+      res.status(401).json({
+        status: res.statusCode,
+        error: 'Invalid credentials',
+      });
+    }
```

Comment:
@T2Wil is there any particular reason for repeating these lines of code?

Reply:
@mcaleb808 , I am checking email first and if matches then it checks the password. 
Password checking block is nested inside the email checking block.
As I am having two 'if blocks' then I am going to need an else block
for each to return error if conditions not met

Reply:
@mcaleb808, Or the other approach is to leave out the else statements and whenever it reaches the end of the block no returned response then I return the error message.
I think this could be the solution.
I am going to implement it.

Reply:
@mcaleb808, Solved!